packages/plugin-restify/src/restify.js

```diff
@@ -27,7 +27,7 @@ module.exports = {
       requestClient.addOnError((event) => {
         const { request, metadata } = getRequestAndMetadataFromReq(req)
         event.request = { ...event.request, ...request }
-        requestClient.addMetadata('request', metadata)
+        event.addMetadata('request', metadata)
```

Comment:
are there any side effects of removing the metadata from the requestClient?

Reply:
Hmm, yes actually though I'm not sure it could be a problem in practice

The way it worked before would mean users could get this data from the request client after an error had been reported, which wouldn't work anymore

Reply:
Given this is a bug it seems reasonable to change the behaviour here (without a significant version bump) as anyone doing this would be relying on a bug

Reply:
Yeah, especially because even if someone was relying on this they'd already have to handle the metadata being missing sometimes